hail/python/test/hail/genetics/test_reference_genome.py

```diff
@@ -136,3 +136,31 @@ def test_liftover_strand(self):
             hl.eval(hl.liftover(hl.parse_locus_interval('1:10000-10000', reference_genome='GRCh37'), 'GRCh38'))
 
         grch37.remove_liftover("GRCh38")
+
+    def test_read_custom_reference_genome(self):
+        # this test doesn't behave properly if these reference genomes are already defined in scope.
+        available_rgs = set(hl.ReferenceGenome._references.keys())
+        self.assertTrue('test_rg_0' not in available_rgs)
+        self.assertTrue('test_rg_1' not in available_rgs)
+        self.assertTrue('test_rg_2' not in available_rgs)
+
+        def assert_rg_loaded_correctly(name):
+            rg = hl.get_reference(name)
+            self.assertEqual(rg.contigs, ["1", "X", "Y", "MT"])
+            self.assertEqual(rg.lengths, {"1": 5, "X": 4, "Y": 3, "MT": 2})
+            self.assertEqual(rg.x_contigs, ["X"])
```

Comment:
Also, this test doesn't seem to be the same as the bug in the original issue you opened.

Reply:
This is the same bug---reading in a dataset that contains a reference genome that the current hail context doesn't already know about is what triggers the bug. (In the issue, the first python snippet generates the problematic dataset; the second python snippet is what actually triggers the bug when trying to read.)

Reply:
(the necessary part of the test that addresses the bug is the part that calls `read_table`; the rest double-checks that the reference genome gets passed correctly into python and does the right thing with inconsistent reference genomes with the same name.)